### ISSUE TITLE
h5py installation error with custom HDF5 path

### ISSUE THREAD

**Issue by AI-Pranto** [h5py/h5py#1566]
During the installation of h5py with the custom path of HDF5, I end up with the following error

At top level:

```
    /home/pranto/.local/lib/python3.6/site-packages/numpy/core/include/numpy/__multiarray_api.h:1463:1: warning: ‘_import_array’ defined but not used [-Wunused-function]
     _import_array(void)
     ^~~~~~~~~~~~~

Command "/usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-pbfts7ym/h5py/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-at7kb5gh-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-pbfts7ym/h5py/
```

**Input command :** ```HDF5_DIR=$HOME/Serial_hdf5/ CC=h5cc pip3 install --no-binary=h5py h5py```

 * Operating System (Ubuntu 18.04 LTS)
 * Python version (3.6)
 * Where Python was acquired (system Python Linux)
 * HDF5 version (1.12.0)

How can I fix this error?

**Comment by keller-mark** [comment]
I am having this same issue, trying to build against hdf5 1.12.0 with --enable-parallel and openmpi 3.1.0

Command
```sh
CC="mpicc" HDF5_MPI="ON" HDF5_DIR=/n/app/hdf5/1.12.0.parallel pip install --no-binary=h5py h5py
```

Output
```
Collecting h5py
  Using cached h5py-2.10.0.tar.gz (301 kB)
Requirement already satisfied: numpy>=1.7 in /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/sit$
-packages (from h5py) (1.18.5)
Requirement already satisfied: six in /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-packa$
es (from h5py) (1.15.0)
Skipping wheel build for h5py, due to binaries being disabled for it.
Installing collected packages: h5py
    Running setup.py install for h5py ... error
    ERROR: Command errored out with exit status 1:
     command: /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/bin/python -u -c 'import sys, setuptools, tokeni$
e; sys.argv[0] = '"'"'/tmp/pip-install-fpkqbpgw/h5py/setup.py'"'"'; __file__='"'"'/tmp/pip-install-fpkqbpgw/h5py/setup.py'"'$
';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(com$
ile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-ji5v1cui/install-record.txt --single-version-externall
y-managed --compile --install-headers /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/include/python3.7m/h5py
         cwd: /tmp/pip-install-fpkqbpgw/h5py/
    Complete output (1444 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/h5py
    copying h5py/__init__.py -> build/lib.linux-x86_64-3.7/h5py

    ...

    copying h5py/tests/test_vds/test_virtual_source.py -> build/lib.linux-x86_64-3.7/h5py/tests/test_vds
    running build_ext
    Loading library to get version: /n/app/hdf5/1.12.0.parallel/lib/libhdf5.so
    Autodetected HDF5 1.12.0
    ********************************************************************************
                           Summary of the h5py configuration

        Path to HDF5: '/n/app/hdf5/1.12.0.parallel'
        HDF5 Version: '1.12.0'
         MPI Enabled: True
    Rebuild Required: True

    ********************************************************************************
    Executing api_gen rebuild of defs
    Executing cythonize()
    [ 1/23] Cythonizing /tmp/pip-install-fpkqbpgw/h5py/h5py/_conv.pyx
    warning: h5py/api_types_hdf5.pxd:418:2: 'H5D_layout_t' redeclared
    warning: h5py/api_types_hdf5.pxd:425:2: 'H5D_alloc_time_t' redeclared
    warning: h5py/api_types_hdf5.pxd:432:2: 'H5D_space_status_t' redeclared
    warning: h5py/api_types_hdf5.pxd:438:2: 'H5D_fill_time_t' redeclared
    warning: h5py/api_types_hdf5.pxd:444:2: 'H5D_fill_value_t' redeclared

    ...

    warning: h5py/api_types_hdf5.pxd:705:2: 'H5Z_filter_t' redeclared
    warning: h5py/api_types_hdf5.pxd:740:6: 'H5Z_ERROR_EDC' redeclared
    warning: h5py/api_types_hdf5.pxd:741:6: 'H5Z_DISABLE_EDC' redeclared
    warning: h5py/api_types_hdf5.pxd:742:6: 'H5Z_ENABLE_EDC' redeclared
    warning: h5py/api_types_hdf5.pxd:743:6: 'H5Z_NO_EDC' redeclared
    building 'h5py.defs' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/tmp
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-fpkqbpgw
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-fpkqbpgw/h5py
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-fpkqbpgw/h5py/h5py
    mpicc -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DH5_USE_16_API -I./h5py -I/n/app/hdf5/1.12.
0.parallel/include -I/tmp/pip-install-fpkqbpgw/h5py/lzf -I/home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/p
ython3.7/site-packages/numpy/core/include -I/home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-
packages/mpi4py/include -I/home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/include/python3.7m -c /tmp/pip-instal
l-fpkqbpgw/h5py/h5py/defs.c -o build/temp.linux-x86_64-3.7/tmp/pip-install-fpkqbpgw/h5py/h5py/defs.o
    In file included from /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-packages/numpy/cor
e/include/numpy/ndarraytypes.h:1832:0,
                     from /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-packages/numpy/cor
e/include/numpy/ndarrayobject.h:12,

    ...

H5_DLL herr_t H5Sencode2(hid_t obj_id, void *buf, size_t *nalloc, hid_t fapl);
                   ^~~~~~~~~~
    In file included from /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-packages/numpy/cor
e/include/numpy/ndarrayobject.h:21:0,
                     from /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-packages/numpy/cor
e/include/numpy/arrayobject.h:4,
                     from /tmp/pip-install-fpkqbpgw/h5py/h5py/api_compat.h:26,
                     from /tmp/pip-install-fpkqbpgw/h5py/h5py/defs.c:652:
    At top level:
    /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/lib/python3.7/site-packages/numpy/core/include/numpy/__mult
iarray_api.h:1463:1: warning: ‘_import_array’ defined but not used [-Wunused-function]
     _import_array(void)
     ^~~~~~~~~~~~~
    error: command 'mpicc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/mk596/software/miniconda3/envs/bigwig-multivec-parallel/bin/python -u -c
 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-fpkqbpgw/h5py/setup.py'"'"'; __file__='"'"'/tmp/pip-i
nstall-fpkqbpgw/h5py/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"',
'"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-ji5v1cui/install-reco
rd.txt --single-version-externally-managed --compile --install-headers /home/mk596/software/miniconda3/envs/bigwig-multivec-p
arallel/include/python3.7m/h5py Check the logs for full command output.
```

**Comment by takluyver** [comment]
The released h5py doesn't work with HDF5 1.12 yet. You either need to use an older version of HDF5, or install h5py from master.
